src/algebraic_geometry/structure_sheaf.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin, Scott Morrison
+-/
+import algebraic_geometry.prime_spectrum
+import algebra.category.CommRing
+import topology.sheaves.local_predicate
+import topology.sheaves.forget
+import ring_theory.localization
+import ring_theory.subring
+
+/-!
+# The structure sheaf on `prime_spectrum R`.
+
+We define the structure sheaf on `Top.of (prime_spectrum R)`, for a commutative ring `R`.
+We define this as a subsheaf of the sheaf of dependent functions into the localizations,
+cut out by the condition that the function must be locally equal to a ratio of elements of `R`.
+
+Because the condition "is equal to a fraction" passes to smaller open subsets,
+the subset of functions satisfying this condition is automatically a subpresheaf.
+Because the condition "is locally equal to a fraction" is local,
+it is also a subsheaf.
```

Comment:
Yes? And that's what we're doing. It seems from your comment that I should improve the documentation some. Maybe just add pointers back to `topology.sheaves.presheaf_of_functions`, `topology.sheaves.sheaf_of_functions`, and `topology.sheaves.local_predicate`?

Reply:
I added

```
(It may be helpful to refer back to `topology.sheaves.sheaf_of_functions`,
where we show that dependent functions into any type family form a sheaf,
and also `topology.sheaves.local_predicate`, where we characterise the predicates
which pick out sub-presheaves and sub-sheaves of these sheaves.)
```
to the module-doc.